app/Jobs/ElasticSearchIndexInit.php

```diff
@@ -1,49 +1,82 @@
 <?php
 
 namespace App\Jobs;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use App\WikiSetting;
+use App\Http\Curl\CurlRequest;
+use App\Http\Curl\HttpRequest;
+use Illuminate\Support\Facades\Log;
 
-class ElasticSearchIndexInit extends Job
+class ElasticSearchIndexInit extends Job implements ShouldBeUnique
 {
     private $wikiDomain;
+    private $wikiId;
+
+    private $request;
 
     /**
      * @return void
      */
-    public function __construct($wikiDomain)
+    public function __construct( string $wikiDomain, int $wikiId, HttpRequest $request = null )
     {
         $this->wikiDomain = $wikiDomain;
+        $this->wikiId = $wikiId;
+        $this->request = $request ?? new CurlRequest();
+    }
+
+    /**
+     * The unique ID of the job.
+     *
+     * @return string
+     */
+    public function uniqueId()
+    {
+        return $this->wikiDomain;
     }
 
     /**
      * @return void
      */
     public function handle()
     {
-        $curl = curl_init();
-        curl_setopt_array($curl, [
-            CURLOPT_URL => getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php?action=wbstackElasticSearchInit&format=json',
-            CURLOPT_RETURNTRANSFER => true,
-            CURLOPT_ENCODING => '',
-            CURLOPT_TIMEOUT => 10,
-            CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
-            CURLOPT_CUSTOMREQUEST => 'POST',
-            CURLOPT_HTTPHEADER => [
-                'content-type: application/x-www-form-urlencoded',
-                'host: '.$this->wikiDomain,
-            ],
-        ]);
-
-        $rawResponse = curl_exec($curl);
-        $err = curl_error($curl);
+        $setting = WikiSetting::where([ 'wiki_id' => $this->wikiId, 'name' => WikiSetting::wwExtEnableElasticSearch, ])->first();
+
+        // job got triggered but no setting in database
+        if ( $setting === null ) {
+            $this->fail(
+                new \RuntimeException('wbstackElasticSearchInit call for '.$this->wikiDomain.' was triggered but not setting available.')
+            );
+
+            return;
+        }
+        
+        $this->request->setOptions( 
+            [
+                CURLOPT_URL => getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php?action=wbstackElasticSearchInit&format=json',
```

Comment:
If the setting is false, then won't this API call fail? as elasticsearch will not be enabled?
Or am I missing something?

Reply:
Do you think it would be ok to enable it on creation and have the job turn it off it it fails?

This would leave the site without working search until the job has either completed or failed.

Reply:
I guess we could enable / disable it in the job too but not sure if it would be necessary

Reply:
> Do you think it would be ok to enable it on creation and have the job turn it off if it fails?
> 
> This would leave the site without working search until the job has either completed or failed.

I think this should be fine for now.